fetch/client/src/main/scala/endpoints4s/fetch/ChunkedEntities.scala

```diff
@@ -4,14 +4,70 @@ import endpoints4s.algebra
 import org.scalajs.dom
 
 import scala.scalajs.js
+import scala.scalajs.js.JSConverters._
+import scala.scalajs.js.PropertyDescriptor
 import scala.scalajs.js.typedarray.Uint8Array
 import scala.scalajs.js.|
 
+trait ChunkedEntities extends ChunkedRequestEntities with ChunkedResponseEntities
+
+trait ChunkedRequestEntities
+    extends algebra.ChunkedRequestEntities
+    with EndpointsWithCustomErrors
+    with Chunks {
+
+  def textChunksRequest: RequestEntity[Chunks[String]] =
+    chunkedRequestEntity(string => new TextEncoder("utf-8").encode(string))
+
+  def bytesChunksRequest: RequestEntity[Chunks[Array[Byte]]] =
+    chunkedRequestEntity(byteArray => Uint8Array.from(byteArray.map(_.toShort).toJSArray))
+
+  private[fetch] def chunkedRequestEntity[A](
+      toUint8Array: A => Uint8Array,
+      framing: dom.ReadableStream[Uint8Array] => dom.ReadableStream[Uint8Array] = identity
+  ): RequestEntity[Chunks[A]] = { (as, request) =>
+    val readableStream = dom.ReadableStream[Uint8Array](
+      new dom.ReadableStreamUnderlyingSource[Uint8Array] {
+        start = js.defined((controller: dom.ReadableStreamController[Uint8Array]) => {
+          def read(reader: dom.ReadableStreamReader[A]): js.Promise[Unit] = {
+            reader
+              .read()
+              .`then`((chunk: dom.Chunk[A]) => {
+                if (chunk.done) {
+                  controller.close(): Unit | js.Thenable[Unit]
+                } else {
+                  controller.enqueue(toUint8Array(chunk.value))
+                  read(reader): Unit | js.Thenable[Unit]
+                }
+              })
+          }
+
+          read(as.getReader()): js.UndefOr[js.Promise[Unit]]
+        }): js.UndefOr[
+          js.Function1[dom.ReadableStreamController[Uint8Array], js.UndefOr[js.Promise[Unit]]]
+        ]
+      }
+    )
+    request.body = framing(readableStream)
+    // https://developer.chrome.com/articles/fetch-streaming-requests/#half-duplex
+    js.Object.defineProperty(
+      request,
+      "duplex",
+      new PropertyDescriptor {
+        configurable = true
+        enumerable = true
+        value = "half"
+        writable = true
+      }
+    )
```

Comment:
Better solution is needed for this. As per https://developer.chrome.com/articles/fetch-streaming-requests/#half-duplex, this will likely end up being browser specific, so maybe we should expose this setting on interpreter level? Settings do not seem like a good spot for the setting because endpoints4s users not using chunked entities should not be forced to deal with it. @julienrf Do you have any suggestions?